tests/nlu/selectors/test_selectors.py

```diff
@@ -207,166 +233,242 @@ def test_resolve_intent_response_key_from_label(
         response_selector.responses[
             util.intent_response_key_to_template_key(label_intent_response_key)
         ]
-        == training_data.responses[
+        == response_selector_training_data.responses[
             util.intent_response_key_to_template_key(resolved_intent_response_key)
         ]
     )
 
 
-async def test_train_model_checkpointing(
-    component_builder: ComponentBuilder, tmpdir: Path
+def test_train_model_checkpointing(
```

Comment:
Uncertain if this adaptation captures the initial test intention 😬

Reply:
So I think the intention here is that when checkpointing is enabled, it seems to write out checkpoint files to a temporary directory and then move them to the model storage later into a 'checkpoints' directory (from looking at `DIETClassifierGraphComponent`), and this tests that that happens. 

So I'm not sure it's sufficient to check that just the model is loaded correctly, you'd need to check that the directory exists and has some stuff in it (though the original comments seem to suggest the number of things in there is a bit variable so you might want to look at that for inspiration). I'm not sure if this is necessary if the underlying class tests it, but I'm not sure I have enough context on that.

Reply:
I think it'd be great to inspect the `model_storage` whether it contains the checkpoint files 👍🏻

Reply:
I modified the test to resemble the one for `DIETClassifier` - however, the `checkpoints` directory doesn't exist, but the files are there 🤔 . I've commented out the 2 lines that fail in the test.

Reply:
I found why that part of the test fails but not how to fix it though. The part of custom config I pass when creating the response selector (e.g. CHECKPOINT_MODEL default set to `True`) doesn't overwrite the default config for some reason that I yet don't understand 🤔

Reply:
instead of doing files > 4 you just wanna check if the `checkpoint` dir exists, right? And that exists for me when running the test 🤔

Reply:
https://github.com/RasaHQ/rasa/blob/f78a6a17d5146248500630a2397ee099a24e0f5c/rasa/utils/train_utils.py#L201

Reply:
Ahaa, this test config needed settings for `EVAL_NUM_EXAMPLES` & `EVAL_NUM_EPOCHS` to pass. All amended now 🎉